File: core/src/main/java/com/a65apps/core/interactors/persons/PersonListModel.java
package com.a65apps.core.interactors.persons;

import com.a65apps.core.entities.Person;

import java.util.List;

import io.reactivex.rxjava3.core.Single;

public class PersonListModel implements PersonListInteractor {
    private final PersonListRepository repository;

    public PersonListModel(PersonListRepository repository) {
        this.repository = repository;
    }

    @Override
    public Single<List<Person>> loadAllPersons(final String searchString) {
        return this.repository.getAllPersons(searchString);
    }
}

File: core/src/main/java/com/a65apps/core/interactors/persons/PersonListModel.kt
package com.a65apps.core.interactors.persons

import com.a65apps.core.entities.Person
import kotlinx.coroutines.flow.Flow

class PersonListModel(private val repository: PersonListRepository) : PersonListInteractor {
    override fun loadAllPersonsFlow(searchString: String): Flow<List<Person>> {
        return repository.getAllPersonsFlow(searchString)
    }

    override suspend fun loadAllPersons(searchString: String): List<Person> {
        return repository.getAllPersons(searchString)
    }

}